src/webob/acceptparse.py

```diff
@@ -268,6 +678,1197 @@ def _match(self, mask, item):
+           best match. The algorithm has many issues, and does not conform to
+           :rfc:`RFC 7231 <7231>`.
+
+           What should be considered a match depends on the needs of your
+           application (for example, should a language range in the header
+           match a more specific language tag offer, or a less specific tag
+           offer?) :rfc:`RFC 7231, section 5.3.5 <7231#section-5.3.5>` suggests
+           three matching schemes from :rfc:`RFC 4647 <4647>`, two of which
+           WebOb supports with
+           :meth:`AcceptLanguageValidHeader.basic_filtering` and
+           :meth:`AcceptLanguageValidHeader.lookup` (we interpret the RFC to
+           mean that Extended Filtering cannot apply for the
+           ``Accept-Language`` header, as the header only accepts basic
+           language ranges.) :meth:`AcceptLanguageValidHeader.basic_filtering`
+           returns quality values with the matched language tags.
+           :meth:`AcceptLanguageValidHeader.lookup` returns a language tag
+           without the quality value, but the quality value is less likely to
+           be useful when we are looking for a best match.
+
+           If these are not suitable or sufficient for the needs of your
+           application, you may need to write your own matching using
+           :attr:`AcceptLanguageValidHeader.parsed`.
+
+        :param offer: (``str``) language tag offer
+        :return: (``float`` or ``None``)
+
+                 | The highest quality value from the language range(s) that
+                   match the `offer`, or ``None`` if there is no match.
+
+
+        **Issues**:
+
+        - Incorrect handling of ``q=0`` and ``*``::
+
+              >>> header = AcceptLanguageValidHeader(header_value='en;q=0, *')
+              >>> header.quality(offer='en')
+              1.0
+
+        - Matching only takes into account the first subtag when matching a
+          range with more specific or less specific tags::
+
+              >>> header = AcceptLanguageValidHeader(header_value='zh')
+              >>> header.quality(offer='zh-Hans-CN')
+              1.0
+              >>> header = AcceptLanguageValidHeader(header_value='zh-Hans')
+              >>> header.quality(offer='zh-Hans-CN')
+              >>> header.quality(offer='zh-Hans-CN') is None
+              True
+
+              >>> header = AcceptLanguageValidHeader(header_value='zh-Hans-CN')
+              >>> header.quality(offer='zh')
+              1.0
+              >>> header.quality(offer='zh-Hans')
+              >>> header.quality(offer='zh-Hans') is None
+              True
+
+        """
+        warnings.warn(
+            'The behavior of AcceptLanguageValidHeader.quality is'
+            'currently being maintained for backward compatibility, but it may'
+            ' be deprecated in the future as it does not conform to the RFC.',
+            PendingDeprecationWarning,
```

Comment:
Make this a `DeprecationWarning`. I really don't think it is a good idea to keep this around for very long.